talktorials/1_ChEMBL/T1_ChEMBL.ipynb

```diff
@@ -10,7 +10,7 @@
     "\n",
```

Comment:
<p>Technically is not a list, but a _lazy_ list (generator-like). Data is not there until is requested. That's why the DataFrame construction takes so long some cells below: only when <code>pandas</code> iterates over the "list" is the data being downloaded. If you do <code>bioactivities[0]</code>, the request took place at the precise moment. See that if you do <code>bioactivies[100]</code> it takes longer.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/volkamerlab/TeachOpenCADD/pull/8/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Reply:
<p>Thanks for pointing this out! I changed this to the following:</p><p> </p><p>The results of the query are stored in <code>bioactivities</code>, a <code>QuerySet</code>, i.e. the results are not fetched from ChEMBL unitl we ask for it (here using <code>pandas.DataFrame.from_records</code>).</p><p> </p><p>More information about the <code>QuerySet</code> datatype:</p><p> </p><p>&gt; QuerySets are lazy – the act of creating a QuerySet doesn’t involve any database activity. You can stack filters together all day long, and Django won’t actually run the query until the QuerySet is evaluated. </p><p>From: https://docs.djangoproject.com/en/3.0/topics/db/queries/#querysets-are-lazy</p>